File: api/src/endpoint/question/get_questions.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::question::QuestionResource};

pub struct GetQuestions;

impl Endpoint for GetQuestions {
    const PATH: &'static str = "/repository/{owner}/{repo}/questions";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetQuestionsRequest;
    type Response = GetQuestionsResponse;
}

#[derive(Debug, Serialize, Deserialize)]
pub struct GetQuestionsRequest;

pub type GetQuestionsResponse = Vec<QuestionResource>;

File: gitdot-api-ts/src/endpoint/question/get-questions.ts
import { z } from "zod";
import { QuestionResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const GetQuestionsRequest = z.object({});
export type GetQuestionsRequest = z.infer<typeof GetQuestionsRequest>;

export const GetQuestionsResponse = z.array(QuestionResource);
export type GetQuestionsResponse = z.infer<typeof GetQuestionsResponse>;

export const GetQuestions = {
  path: "/repository/{owner}/{repo}/questions",
  method: "GET",
  request: GetQuestionsRequest,
  response: GetQuestionsResponse,
} satisfies Endpoint;
export type GetQuestions = typeof GetQuestions;
